src/ray/raylet/scheduling_queue.cc

```diff
@@ -69,7 +69,13 @@ bool TaskQueue::AppendTask(const TaskID &task_id, const Task &task) {
   task_map_[task_id] = list_iterator;
   // Resource bookkeeping
   total_resource_load_.AddResources(task.GetTaskSpecification().GetRequiredResources());
-  resource_load_by_shape_[task.GetTaskSpecification().GetSchedulingClass()]++;
+  const auto &scheduling_class = task.GetTaskSpecification().GetSchedulingClass();
+  resource_load_by_shape_[scheduling_class]++;
+
+  int64_t backlog_size = task.BacklogSize();
+  if (backlog_size >= 0) { // Poor man's version of RayConfig::instance().report_worker_backlog()
```

Comment:
I haven't seen any other calls to `RayConfig::instance()` on the critical path, so I'm assuming it's too expensive to do here.

Reply:
Yeah I think accessing global field is pretty expensive. Not sure if it matters in raylet though.